lib/cms/repo.ex

```diff
@@ -130,7 +131,28 @@ defmodule CMS.Repo do
   def do_banner do
     case @cms_api.view("/cms/important-notices", []) do
       {:ok, [api_data | _]} -> Banner.from_api(api_data)
-      {:ok, _} -> :empty
+      {:ok, _} -> nil
+      {:error, _} -> :error
+    end
+  end
+
+  @spec photo() :: Photo.t() | nil
+  def photo do
+    cached_value = do_photo()
+
+    if cached_value == :empty || cached_value == :error, do: nil, else: cached_value
+  end
+
+  @decorate cacheable(
+              cache: @cache,
+              key: "cms.repo|photos",
+              on_error: :nothing,
+              opts: [ttl: @ttl]
+            )
+  def do_photo do
+    case @cms_api.view("/cms/photos", []) do
+      {:ok, [api_data | _]} -> Photo.from_api(api_data)
+      {:ok, _} -> nil
       {:error, _} -> :error
```

Comment:
I feel like there is a better way to handle this. We are using :error and :empty and nil when I think we can limit it to two different responses. Either {:ok, value} or {:error, reason}.